### PR TITLE
Configurable sink

### PR DESCRIPTION
**Pull request by whatyouhide**
Hey @ferd!

We're opening a new PR (in place of #10) that includes all the commits that were in #10 and some more to address the comments you left on #10. Sorry for all the mess with the two PRs, but #10 was based on our `master` and we're using that internally so we didn't want to change it too much :).

Specifically, I addressed these things (on top of what #10 already changed):

- added a `vmstats_sink` behaviour as we talked about (3285955)
- added a `key_separator` option to vmstats's env (defaulting to `$.`, used to separate the parts that compose metric keys)
- removed the `Makefile`
- removed the duplicated `applications` tuple in `vmstats.app.src`
- enabled the scheduler wall time by default and dropped support for `unavailable` (19f00d0)
- added support for `rebar3` (a bunch of new fields to `vmstats.app.src` and a `rebar.lock` file) (2a0ec69
)
- added an entry in the changelog for v2.0.0 and moved the changelog to a separate file (`CHANGELOG.md`)
- added a bunch of new documentation to the README

I also have a couple of questions for you:

1. regarding the `vmstats_statsderl` project: do you want to create it under your GitHub handle, or should we create it under https://github.com/elixir-addicts (giving you access of course)? Either way, we can maintain it or help maintaining it :).
2. would you be willing to push vmstats to Hex? Of course, we can take care of that if you're ok with it.

Let me know how everything looks and if we can do anything else :).

(as always, @lexmag did most of the work here so let's include him in the conversation :smile:)

### CODE REVIEW

**Comment by ferd** [comment]
If you want to maintain the vmstats_statsderl project, you can host it, I'll be fine with it. If it's all moved to hex anyway most users won't see a major difference. 

I could push vmstats to hex once this gets merged in, yes.

Thanks for all the work on this.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+### Configuration
+
+The following is a list of the possible options for the configuration of the
+`vmstats` app:
+
+  * `sink` - (module) a module that implements the `vmstats_sink` behaviour; vmstats metrics will be collected through this module.
+  * `base_key` - (string) every metric name is prepended with this base key. Defaults to `"vmstats"`.
+  * `key_separator` - (char) used as a separator between the parts of metric keys. Defaults to `$.`.
+  * `delay` - (integer) the time (in milliseconds) between metric gatherings. Defaults to `1000` (1s).
```
Comment (lexmag) [inline README.md]: @ferd considering we're doing version change, how do you feel about changing `delay` parameter name to `interval` as well? It feels a bit more self-descriptive to me.

Comment (ferd) [inline README.md]: That's fine yes.

**Comment by whatyouhide** [comment]
@ferd I updated the PR with a couple of new commits (one of them renames `delay` to `interval`, the other ones cleans up the `test/statsderl.erl` file a little bit).

**Comment by ferd** [comment]
@whatyouhide I had this on the backburner. I think this is ready to merge (will need to bump 2.0.0) once the statsderl stuff is handled in a side project (maybe it is and I just don't know yet)

**Comment by whatyouhide** [comment]
@ferd nope, still now handled yet (we were waiting on the verdict :smiley:). Gonna work on it and get back to you once it's ready.

**Comment by whatyouhide** [comment]
Hey @ferd, we created a statsderl sink in the [vmstats_statsderl project](https://github.com/elixir-addicts/vmstats_statsderl). Right now, it uses our elixir-addicts/vmstats fork (this PR basically) so that it's actually usable right now; we'll update to point to the original repo as soon as this PR is merged. Let me know what you think and if we can do anything else.

**Comment by ferd** [comment]
Configured and pushed tag 2.0.0.

**Comment by whatyouhide** [comment]
@ferd awesome :tada:! Now, it would be just great if you could publish this to hex.pm. It should be straighforward ([guide here](https://hex.pm/docs/rebar3_publish)).

**Comment by ferd** [comment]
Yep that's done.

**Comment by lexmag** [comment]
@ferd thank you! :yellow_heart:

**Comment by ferd** [comment]
Well, thanks for all your contributions on this!

**Pull request merged**